source/MRCuda/MRCudaSolarRadiation.cpp

```diff
@@ -0,0 +1,217 @@
+#include "MRCudaSolarRadiation.h"
+#include "MRMesh/MRAABBTree.h"
+#include "MRCudaSolarRadiation.cuh"
+
+namespace MR
+{
+namespace Cuda
+{
+
+static void findMaxVectorDim( int& dimX, int& dimY, int& dimZ, const Vector3f& dir )
```

Comment:
Could we have one
```
template <typename T>
void findMaxVectorDim( int& dimX, int& dimY, int& dimZ, const Vector3<T>& dir )
```
function both for CPU and CUDA without copy-paste?

If necessary, plese move it in a new header file.